src/Microsoft.Diagnostics.Monitoring.Egress/FileSystem/FileSystemEgressEndpoint.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.Egress.FileSystem
+{
+    internal class FileSystemEgressEndpoint : IEgressEndpoint<FileSystemEgressStreamOptions>
+    {
+        private readonly FileSystemEgressEndpointOptions _endpointOptions;
+
+        public FileSystemEgressEndpoint(FileSystemEgressEndpointOptions endpointOptions)
+        {
+            _endpointOptions = endpointOptions;
+        }
+
+        public async Task<EgressResult> EgressAsync(
+            Func<CancellationToken, Task<Stream>> action,
+            string name,
+            FileSystemEgressStreamOptions streamOptions,
+            CancellationToken token)
+        {
+            using var fileStream = CreateFileStream(name, out string filePath);
+
+            using Stream stream = await action(token);
+            
+            await stream.CopyToAsync(fileStream, 0x1000, token);
```

Comment:
Consider making the buffersize one of the options for the egress

Reply:
Are you suggesting this is a common option on the EgressEndpointOptions class (so it can be set when describing the options for an endpoint e.g. the settings file has a file system endpoint with a custom copy buffer size), or this is more an implementation detail that is passed to the endpoint class? Should this also be used as the file stream buffer size as well?

Reply:
My original suggestion (when CopyToAsync would ideally be moved into the base class), was to make it part of EgressEndpointOptions. 

For file stream buffering, I'm not sure if this is a really meaningful setting for writes.

Reply:
I've added back the EgressEndpointOptions class with a CopyBufferSize option. The endpoints have been updated to have a default implemented pull model that uses the CopyBufferSize to copy from the pulled stream into the stream provided by the push model. The default that the endpoints use if the option is not specified is 1MiB (0x100000 bytes).